source/GameAction.cpp

```diff
@@ -436,8 +456,22 @@ void GameAction::Do(PlayerInfo &player, UI *ui, const Mission *caller) const
 
 	for(const System *system : mark)
```

Comment:
While we're here, can we mark the `mark` and `unmark` for loops in an `if(caller)`? I don't think `caller` can ever be nullptr, but just in case. Line 485 is also checking if `caller` is not null, so maybe just combine these loops with that if statement.